transit_odp/avl/post_publishing_checks/daily/vehicle_journey_finder.py

```diff
@@ -446,7 +445,9 @@ def match_vehicle_activity_to_vehicle_journey(
 
         txc_xml = self.get_corresponding_timetable_xml_files(matching_txc_file_attrs)
 
-        if not self.filter_by_operating_period(activity_date, txc_xml, result):
+        if not self.filter_by_operating_period(
+            activity.recorded_at_time.date(), txc_xml, result
+        ):
```

Comment:
This looks correct for the happy path. However, if `RecordedAtTime` is missing from the incoming packet (because it's malformed), you'll get an exception here when it accesses the `date` member. All the rest of the PPC code defensively checks that elements XML exist before using their content or accessing sub-elements. Can you please add a defensive check here (and below) - either a check on the value of `recorded_at_time` or a `try...except` block, whichever you prefer.